src/main/webapp/app/shared/resizeable-container/resizeable-container.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, AfterViewInit, Input, HostBinding, ElementRef, Renderer2 } from '@angular/core';
+import interact from 'interactjs';
+
+/**
+ * Resizable Layout with collapsable panel on the right hand side.
+ *
+ * Usage Example:
+ * <jhi-resizeable-container>
+ *
+ *    <span left-header>Header Left</span>
+ *    <p left-body>Body Left</p>
+ *
+ *    <fa-icon icon="exclamation-triangle" right-header></fa-icon>
+ *    <span right-header>Header Right</span>
+ *    <p right-body>Body Right</p>
+ *
+ *  </jhi-resizeable-container>
+ */
+@Component({
+    selector: 'jhi-resizeable-container',
+    templateUrl: './resizeable-container.component.html',
+    styleUrls: ['./resizeable-container.component.scss'],
+})
+export class ResizeableContainerComponent implements OnInit, AfterViewInit {
+    @Input() collapsed = false;
+
+    private readonly nativeElement: HTMLElement;
+
+    constructor(elementRef: ElementRef, private render: Renderer2) {
+        this.nativeElement = elementRef.nativeElement;
+    }
+
+    ngOnInit() {
+        this.render.addClass(this.nativeElement, 'flex-grow-1');
```

Comment:
Does adding the class have to happen dynamically?

Reply:
Moved this as a host binding. Good catch.